BioArchLinux/rstudio-desktop/PKGBUILD

```diff
@@ -0,0 +1,138 @@
+            '2c69e7b040c208b61ebf9735c63d2e5bcabfed32ef05a9b8dd5823489ea50d6b'
+            '723626bfe05dafa545e135e8e61a482df111f488583fef155301acc5ecbbf921'
+            '286925c442c1818979714feeec1577f03ae8a3527d2478b0f55238e2272a0b9e')
+
+noextract=("gin-${_ginver}.zip")
+
+# Retrives commit id of git tag associated with release  
+_get_commit_from_tag() {
+  git ls-remote https://github.com/rstudio/rstudio.git | grep v${pkgver} | sed 's/^\(.\{7\}\).*/\1/'
+}
+
+prepare() {
+    cd ${srcdir}/${_srcname}
+    # Do not use outdated version name of pandoc
+    patch -p1 < ${srcdir}/pandoc_version.patch
+
+    cd "${srcdir}/${_srcname}/dependencies/common"
+    install -d pandoc/${_pandocver}
+ 
+    ln -sfT /usr/share/myspell/dicts dictionaries
+    ln -sfT /usr/share/mathjax2 mathjax-27
+    ln -sfT /usr/bin/pandoc pandoc/${_pandocver}/pandoc
+
+    # Nodejs
+    install -d node/${_nodever}
+    cp -r "${srcdir}/node-v${_nodever}-linux-x64/"* node/${_nodever}
+    #cd "${srcdir}/${_srcname}/src/gwt/panmirror/src/editor"
+    # yarn config set ignore-engines true
+    #yarn install
+
+    # Fix links for src/cpp/session/CMakeLists.txt
+    cd "${srcdir}/${_srcname}/dependencies"
+    ln -sfT /usr/share/myspell/dicts dictionaries
+    ln -sfT /usr/share/mathjax2 mathjax-27
+    #ln -sfT /usr/bin/pandoc pandoc
+
+    # Panmirror is picked up now from Quarto repo
+    # Ideally: git clone --branch release/rstudio-cherry-blossom https://github.com/quarto-dev/quarto.git "${srcdir}/${_srcname}/src/gwt/lib/quarto"
+    mkdir -p "${srcdir}/${_srcname}/src/gwt/lib/quarto"
+    cp -r "${srcdir}/quarto-release-rstudio-cherry-blossom/"* "${srcdir}/${_srcname}/src/gwt/lib/quarto"
+}
+
+build() {
+
+    # Quarto
+    msg "Checking if Quarto is installed..."
+
+    if (pacman -Q quarto >/dev/null) ; then
+        msg "Enabling Quarto support..."
+        _quarto="TRUE"
+        cd "${srcdir}/${_srcname}/dependencies"
+        install -d quarto/bin/tools
+        ln -sfT /usr/bin/quarto quarto/bin/quarto
+        ln -sfT /usr/bin/pandoc quarto/bin/tools/pandoc
+    else
+        msg "Use pandoc, because Quarto is not used..."
+        cd "${srcdir}/${_srcname}/dependencies"
+        install -d pandoc/${_pandocver}/bin/tools
+        ln -sfT /usr/bin/pandoc pandoc/${_pandocver}/bin/tools/pandoc
+    fi
+
+    cd ${srcdir}
+
+    
+    # Use path of system provided R packages
+    export R_LIBS_USER=/usr/lib/R/library/
+
+    export PATH=/usr/lib/jvm/java-8-openjdk/jre/bin/:${PATH}
+    # These variables are used to create contents of Help > About dialog 
+    export RSTUDIO_VERSION_MAJOR=${_pkgver%%.*}
+    export RSTUDIO_VERSION_MINOR=${_pkgver#*.}
+    export RSTUDIO_VERSION_PATCH=${_pkgver#*.*.}
+    export RSTUDIO_VERSION_SUFFIX="+${_pkgver##*+}"
+    export GIT_COMMIT=${_gitcommit}
+    export PACKAGE_OS=$(uname -om)
+    
+    cmake -S "${srcdir}/${_srcname}" -B build \
+          -DRSTUDIO_TARGET=Desktop \
```

Comment:
According to the upstream [INSTALL file](https://github.com/rstudio/rstudio/blob/v2023.06.2%2B561/INSTALL#L10), the recommended `RSTUDIO_TARGET` value is `Electron` which builds an electron GUI instead of the legacy Qt GUI.

If this change is too troublesome to do, we can use the legacy Qt GUI for now, but this should be changed at some point.

Reply:
i am familiar with Qt build of rstudio. would need to someone to test out electron builds.